Fast builds by default (#8925)
ray-project/ray: doc/source/development.rst

This command is not enough to recompile all C++ unit tests. To do so, see
`Testing locally`_.

> Fast, Debug, and Optimized Builds
> ----------------------------

Reviewer: ```suggestion
Fast, Debug, and Optimized Builds
---------------------------------
```

This command is not enough to recompile all C++ unit tests. To do so, see
`Testing locally`_.

Optimized & Debug Builds
----------------------------

> The default build configuration may skip performing compile-time optimizations.

Reviewer 2: The doc needs to be flipped now right?

Author: Yeah I guess, I updated it.

This command is not enough to recompile all C++ unit tests. To do so, see
`Testing locally`_.

Optimized & Debug Builds
----------------------------

The default build configuration may skip performing compile-time optimizations.

> This is to ensure that compilation remain as fast as possible,

Reviewer 3: ```suggestion
This is to ensure that compilation remain as fast as possible
```

and to make debugging easier.
However, this can make Ray slower to run, and hence unsuitable for benchmarking.

> For a optimized build, you can try the following:

Reviewer 3: ```suggestion
For an optimized build, you can do the following:
```

.. code-block:: shell

 bazel build -c opt "//:ray_pkg"

> This will build Ray's Python package with optimizations.
> (Note that this might take a while.)

Reviewer 3: ```suggestion
This will build Ray's Python package with optimizations (this may take a while).
```
